shared/Microsoft.Extensions.ActivatorUtilities.Sources/ActivatorUtilities.cs

```diff
@@ -294,7 +294,7 @@ public object CreateInstance(IServiceProvider provider)
                 {
                     return _constructor.Invoke(_parameterValues);
                 }
-                catch (Exception ex)
+                catch (Exception ex) when (ex.InnerException != null)
```

Comment:
Maybe we should just catch different exception type? Like `TargetInvocationException`?